src/SBitSet.jl

```diff
@@ -0,0 +1,284 @@
+
+
+@propagate_inbounds function MBitSet{N}(idxs::Vector) where N
+    a = zero(MBitSet{N})
+    for idx in idxs
+        a[idx] = true
+    end
+    return a
+end
+@propagate_inbounds SBitSet{N}(idxs::Vector) where N = SBitSet(MBitSet{N}(idxs...))
+
+
+#TODO: Maybe replace loop by branch-free ifelse code
+@propagate_inbounds function MBitSet{N}(idxs::Base.OneTo{<:Integer}) where N
+    a = zero(MBitSet{N})
+    idxs.stop < 1 && return a
+    _checkbounds(a, idxs.stop)
+    i1, i2 = Base.get_chunks_id(idxs.stop)
+    ptr = convert(Ptr{UInt64}, pointer_from_objref(a))
+    GC.@preserve a begin
+        @inbounds for i = 1:i1-1
+            unsafe_store!(ptr, 0xffffffffffffffff, i)
+        end
+        unsafe_store!(ptr, (0xffffffffffffffff>> ((63-i2) & 63)), i1)
+    end
+    return a
+end
+
+@propagate_inbounds SBitSet{N}(idxs::Base.OneTo{<:Integer}) where N = SBitSet(MBitSet{N}(idxs))
+
+@propagate_inbounds function MBitSet{N}(idxs::Base.UnitRange{<:Integer}) where N
+    idxs.start == 1 && return MBitSet{N}(Base.OneTo(idxs.stop))
+    @boundscheck begin 
+         (0 < idxs.start && idxs.stop <= 64*N) || length(idxs)==0 || throw(BoundsError(MBitSet{N}, idxs))
+    end
+    @inbounds return MBitSet{N}(Base.OneTo(idxs.stop)) & ~MBitSet{N}(Base.OneTo(idxs.start-1))
+end
+
+@propagate_inbounds SBitSet{N}(idxs::Base.UnitRange{<:Integer}) where N = SBitSet(MBitSet{N}(idxs))
+
+## Conversion to and from BitVector
+function convert(::Type{BitVector}, a::UBitSet{N})  where N
+    res = BitVector(undef, 64*N)
+    @inbounds for i = 1:N
+        res.chunks[i] = a.chunks[i]
+    end
+    res
+end
+convert(::Type{SBitSet{N}}, a::BitVector) where N = SBitSet(ntuple(i->a.chunks[i], Val{N}()))
+convert(::Type{MBitSet{N}}, a::BitVector) where N = MBitSet(ntuple(i->a.chunks[i], Val{N}()))
+convert(::Type{SBitSet}, a::BitVector) where N = convert(SBitSet{length(a.chunks)}, a)
+convert(::Type{MBitSet}, a::BitVector) where N = convert(MBitSet{length(a.chunks)}, a)
+
+## Equality, Sorting, Hashing
+isequal(L::UBitSet, R::UBitSet) = L.chunks==R.chunks
+==(L::UBitSet, R::UBitSet) where N = L.chunks==R.chunks
+
+#This is taken from Base/hashing2.jl. Make faster, don't rely on unexported internals.
+@inline function hash(x::UBitSet{N}, h::UInt = UInt(0)) where N
+    @inbounds for i=1:N
+        h = xor(h, Base.hash_uint(xor(x.chunks[i], h)))
+    end
+    return h
+end
+
+@inline _blsi(a) = a & -a
+@inline function isless(a::UBitSet{N}, b::UBitSet{N}) where N
+    @inbounds for i = 1:N
+        ac = a.chunks[i]
+        bc = b.chunks[i]
+        ac == bc && continue
+        msk = _blsi(xor(ac, bc))
+        return iszero(bc & msk)
+    end
+    return false
+end
+
+## Random
+@inline rand(rng::Random.AbstractRNG, ::Random.SamplerType{SBitSet{N}}) where {N} = SBitSet(ntuple(i->rand(rng, UInt64), Val{N}()))
+
+
+## SIMD opt out. Make less ugly once github.com/JuliaLang/julia/pull/31113 has aged.
```

Comment:
The entire simd opt-out is super fishy. It is also liable to break on new julia versions, since it relies on unexported internals, and the 1.0 vs 1.1-1.3 story is not pretty.

The opt-out is necessary when using code bases that assume that their input is @simd-able. The prime example is LightGraphs, that contains many `@simd for node2 in neighbors(g, node1)`, but otherwise requires no random access to `neighbors(g, node)`. Without a pragmatic opt-out, it would be impossible to use lightgraphs algorithms on custom graph types that use sbitset for neighbors.

Comments?